cpp/src/arrow/util/decimal.h

```diff
@@ -80,7 +80,7 @@ class ARROW_EXPORT Decimal128 : public BasicDecimal128 {
     std::pair<Decimal128, Decimal128> result;
     auto dstatus = BasicDecimal128::Divide(divisor, &result.first, &result.second);
     ARROW_RETURN_NOT_OK(ToArrowStatus(dstatus));
-    return std::move(result);
+    return result;
```

Comment:
Please don't mix unrelated changes.
These changes are for https://github.com/apache/arrow/pull/41107 , right?

Reply:
Ah yes sorry about that. Was working off of one branch to try and get a clean nanoarrow build so this slipped in. Will revert